scipy/sparse/_construct.py

```diff
@@ -14,13 +15,21 @@
 from ._sputils import upcast, get_index_dtype, isscalarlike
 
 from ._sparsetools import csr_hstack
-from ._csr import csr_matrix
-from ._csc import csc_matrix
 from ._bsr import bsr_matrix
 from ._coo import coo_matrix
-from ._dia import dia_matrix
+from ._csc import csc_matrix
+from ._csr import csr_matrix
+from ._dia import dia_matrix, dia_array
+from ._dok import dok_matrix
+from ._lil import lil_matrix
 
 from ._base import issparse
+from ._matrix import _array_doc_to_matrix
+
+_fmt_to_spmatrix = {
```

Comment:
I think we can avoid having to do this: since `diags` (and thus `diags_array`) natively outputs DIA format (when `format=None`), we can have the legacy `diags` function call `A = diags_array(..., format=None)` and then unconditionally convert that to spmatrix like: `return dia_matrix(A).asformat(format)`.